### PR TITLE
feat: add json string input to patterns, update rg logic

### PR DESCRIPTION
**Pull request by argeiger**
### Description

This allows the ability to pass in a json string instead of a filename for the architecture manifest.  Also fixes the resource group when use_prefix is defined in map.  

**Check the relevant boxes:**
- [ ] Bug fix (nonbreaking change that fixes an issue)
- [x] New feature (nonbreaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Examples or tests (addition or updates of examples or tests)
- [ ] Documentation update
- [ ] CI related update (pipeline, etc.)

### Checklist

- [ ] If relevant, a test for the change has been added or updated as part of this PR.
- [x] If relevant, documentation for the change has been added or updated as part of this PR.

### Merge

- Merge using "Squash and merge".
- Make sure to use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents. The commit message determines whether a new version of the modules needs to be released, and if so, which semver number to use).

### CODE REVIEW

**Review by vburckhardt — APPROVED** [review]

File: patterns/mixed/variables.tf [hunk 1]
```diff
@@ -639,4 +639,10 @@ variable "override" {
   default     = false
 }
 
+variable "override_json_string" {
```
Comment (vburckhardt) [inline patterns/mixed/variables.tf:642]: We probably want to have an example using the override_json_string input, and associated tests eventually. I don't think it's blocking this PR tough.

**Review by vburckhardt — APPROVED** [review]

**Comment by ocofaigh** [comment]
@argeiger Pipeline is hitting this one again. I think a workaround should be added like we discussed to make the module more robust to backend slowness issues...

```
│ Error: ServiceNotAuthorized: The specified COS Service Instance does not have sufficient permissions to access the resource associated with the KMS key CRN.
│ 	status code: 401, request id: 6f8b3c41-939d-4ce9-8d74-42864d0d7e6d, host id: 
│ 
│   with module.landing_zone.module.landing_zone.ibm_cos_bucket.buckets["workload-bucket"],
│   on ../../cos.tf line 74, in resource "ibm_cos_bucket" "buckets":
│   74: resource "ibm_cos_bucket" "buckets" {
│ 
╵
╷
│ Error: ServiceNotAuthorized: The specified COS Service Instance does not have sufficient permissions to access the resource associated with the KMS key CRN.
│ 	status code: 401, request id: 1365c280-f65e-46e3-97d2-9dd8b5dd1fc9, host id: 
│ 
│   with module.landing_zone.module.landing_zone.ibm_cos_bucket.buckets["management-bucket"],
│   on ../../cos.tf line 74, in resource "ibm_cos_bucket" "buckets":
│   74: resource "ibm_cos_bucket" "buckets" {
│ 
╵
╷
│ Error: ServiceNotAuthorized: The specified COS Service Instance does not have sufficient permissions to access the resource associated with the KMS key CRN.
│ 	status code: 401, request id: d0008e84-cc02-44e7-a401-ff86cfac130b, host id: 
│ 
│   with module.landing_zone.module.landing_zone.ibm_cos_bucket.buckets["atracker-bucket"],
│   on ../../cos.tf line 74, in resource "ibm_cos_bucket" "buckets":
│   74: resource "ibm_cos_bucket" "buckets" {
│ 
╵}
```

**Comment by vburckhardt** [comment]
@ocofaigh - Do we have a ticket to track this issue?

**Comment by ocofaigh** [comment]
Internal issue has been created to track the issue

**Pull request merged**